docker/server/entrypoint.sh

```diff
@@ -3,164 +3,215 @@
 set -eo pipefail
 shopt -s nullglob
 
-DO_CHOWN=1
-if [ "${CLICKHOUSE_DO_NOT_CHOWN:-0}" = "1" ]; then
-    DO_CHOWN=0
-fi
-
-CLICKHOUSE_UID="${CLICKHOUSE_UID:-"$(id -u clickhouse)"}"
-CLICKHOUSE_GID="${CLICKHOUSE_GID:-"$(id -g clickhouse)"}"
-
-# support --user
-if [ "$(id -u)" = "0" ]; then
-    USER=$CLICKHOUSE_UID
-    GROUP=$CLICKHOUSE_GID
-    if command -v gosu &> /dev/null; then
-        gosu="gosu $USER:$GROUP"
-    elif command -v su-exec &> /dev/null; then
-        gosu="su-exec $USER:$GROUP"
+# check to see if this file is being run or sourced from another script
+_is_sourced() {
+	# https://unix.stackexchange.com/a/215279
+	[ "${#FUNCNAME[@]}" -ge 2 ] \
+		&& [ "${FUNCNAME[0]}" = '_is_sourced' ] \
+		&& [ "${FUNCNAME[1]}" = 'source' ]
+}
+
+get_variables() {
+    declare -g CLICKHOUSE_DO_NOT_CHOWN INIT_ON_EVERY_START
+    CLICKHOUSE_DO_NOT_CHOWN="${CLICKHOUSE_DO_NOT_CHOWN:-1}"
+    # Backwards compatibility
+    INIT_ON_EVERY_START="${INIT_ON_EVERY_START:-1}"
+
+    declare -g CLICKHOUSE_UID CLICKHOUSE_GID
+    CLICKHOUSE_UID="${CLICKHOUSE_UID:-"$(id -u clickhouse)"}"
+    CLICKHOUSE_GID="${CLICKHOUSE_GID:-"$(id -g clickhouse)"}"
+
+    declare -g gosu USER GROUP
+    # support --user
+    if [ "$(id -u)" = "0" ]; then
+        USER=$CLICKHOUSE_UID
+        GROUP=$CLICKHOUSE_GID
+        if command -v gosu &> /dev/null; then
+            gosu="gosu $USER:$GROUP"
+        elif command -v su-exec &> /dev/null; then
+            gosu="su-exec $USER:$GROUP"
+        else
+            echo "No gosu/su-exec detected!"
+            exit 1
+        fi
     else
-        echo "No gosu/su-exec detected!"
-        exit 1
+        USER="$(id -u)"
+        GROUP="$(id -g)"
+        gosu=""
+        CLICKHOUSE_DO_NOT_CHOWN=0
```

Comment:
the meaning of flag is opposite

Reply:
Changed CLICKHOUSE_DO_NOT_CHOWN default variable (back to the original value). Now if the container starts with an arbitrary uid this variable is set to 1